### ISSUE TITLE
Unable to Deploy to Heroku

### ISSUE THREAD

**Issue by lhvy** [Lattyware/massivedecks#176]
## Describe the bug
When deploying to Heroku, the client successfully builds but the server fails to build. Here is the important part of the log:
```
-----> Build
       Running build
       
       > massivedecks@2.0.0 build /tmp/build_a9709bdd_
       > cp -r server/decks decks && npm run buildServer && npm run buildClient
       
       
       > massivedecks@2.0.0 buildServer /tmp/build_a9709bdd_
       > npm explore @massivedecks/server -- npm run build
       
       
       > @massivedecks/server@2.0.0 build /tmp/build_a9709bdd_/server
       > npx tsc
       
npx: installed 1 in 1.385s
       error TS6047: Argument for '--target' option must be 'ES3', 'ES5', or 'ES6'.
       error TS6046: Argument for '--module' option must be 'commonjs', 'amd', 'system' or 'umd'.
       error TS5023: Unknown compiler option 'incremental'.
       error TS5023: Unknown compiler option 'resolveJsonModule'.
       error TS5023: Unknown compiler option 'strict'.
       error TS5023: Unknown compiler option 'strictNullChecks'.
       error TS5023: Unknown compiler option 'strictFunctionTypes'.
       error TS5023: Unknown compiler option 'strictBindCallApply'.
       error TS5023: Unknown compiler option 'strictPropertyInitialization'.
       error TS5023: Unknown compiler option 'noImplicitThis'.
       error TS5023: Unknown compiler option 'alwaysStrict'.
       error TS5023: Unknown compiler option 'noImplicitReturns'.
       error TS5023: Unknown compiler option 'noFallthroughCasesInSwitch'.
       error TS5023: Unknown compiler option 'moduleResolution'.
       error TS5023: Unknown compiler option 'allowSyntheticDefaultImports'.
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @massivedecks/server@2.0.0 build: `npx tsc`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @massivedecks/server@2.0.0 build script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

## How to reproduce the bug
Steps to reproduce the behaviour:
1. Use the Heroku deploy button on this repo's README.md
2. Pick an application name and press the "Deploy app" button on Heroku.

### Expected behaviour
When running a dev environment or building locally, everything works fine. The expected behaviour is that deploying to Heroku works successfully as it does on other platforms.

**Comment by Lattyware** [comment]
Ah, yeah, for some reason it seems to be pulling down an ancient version of the typescript compiler rather than the version we have listed—I'm guessing the two subprojects aren't getting their `node_modules` populated like they used to, for some reason.

**Comment by Yash-Git-Hub** [comment]
> Ah, yeah, for some reason it seems to be pulling down an ancient version of the typescript compiler rather than the version we have listed—I'm guessing the two subprojects aren't getting their `node_modules` populated like they used to, for some reason.

So what do we need to fi I am fairly new to web dev in case you could tell thank you

**Comment by Lattyware** [comment]
Sorry, I haven't had a chance to work out what is going on and the best way to fix it, for now the Heroku build is just broken unless someone can get to the bottom of why it isn't working any more.

**Comment by CookieCoder15** [comment]
Ok so I did some testing, by changing typescript to a production dependency in the `server/package.json`, I was able to make some progress in the right direction and have attached the log below. It now complains that there are no declaration files for a bunch of modules. I tried to move the other dev dependencies such as `@types/express` to prod but the build log still said the declaration files were missing.

```
-----> Build
       Running build
       
       > massivedecks@2.0.0 build /tmp/build_9815baf8_
       > cp -r server/decks decks && npm run buildServer && npm run buildClient
       
       
       > massivedecks@2.0.0 buildServer /tmp/build_9815baf8_
       > npm explore @massivedecks/server -- npm run build
       
       
       > @massivedecks/server@2.0.0 build /tmp/build_9815baf8_/server
       > npx tsc
       
       ../node_modules/express-winston/index.d.ts(7,65): error TS7016: Could not find a declaration file for module 'express'. '/tmp/build_9815baf8_/node_modules/express/index.js' implicitly has an 'any' type.
         Try `npm install @types/express` if it exists or add a new declaration (.d.ts) file containing `declare module 'express';`
       ../node_modules/helmet/dist/index.d.ts(1,49): error TS2307: Cannot find module 'http' or its corresponding type declarations.
       ../node_modules/helmet/dist/middlewares/content-security-policy/index.d.ts(1,23): error TS2688: Cannot find type definition file for 'node'.
       ../node_modules/helmet/dist/middlewares/content-security-policy/index.d.ts(2,49): error TS2307: Cannot find module 'http' or its corresponding type declarations.
       ../node_modules/helmet/dist/middlewares/expect-ct/index.d.ts(1,49): error TS2307: Cannot find module 'http' or its corresponding type declarations.
       ../node_modules/helmet/dist/middlewares/referrer-policy/index.d.ts(1,49): error TS2307: Cannot find module 'http' or its corresponding type declarations.
       ../node_modules/helmet/dist/middlewares/strict-transport-security/index.d.ts(1,49): error TS2307: Cannot find module 'http' or its corresponding type declarations.
       ../node_modules/helmet/dist/middlewares/x-content-type-options/index.d.ts(1,49): error TS2307: Cannot find module 'http' or its corresponding type declarations.
       ../node_modules/helmet/dist/middlewares/x-dns-prefetch-control/index.d.ts(1,49): error TS2307: Cannot find module 'http' or its corresponding type declarations.
       ../node_modules/helmet/dist/middlewares/x-download-options/index.d.ts(1,49): error TS2307: Cannot find module 'http' or its corresponding type declarations.
       ../node_modules/helmet/dist/middlewares/x-frame-options/index.d.ts(1,49): error TS2307: Cannot find module 'http' or its corresponding type declarations.
       ../node_modules/helmet/dist/middlewares/x-permitted-cross-domain-policies/index.d.ts(1,49): error TS2307: Cannot find module 'http' or its corresponding type declarations.
       ../node_modules/helmet/dist/middlewares/x-powered-by/index.d.ts(1,49): error TS2307: Cannot find module 'http' or its corresponding type declarations.
       ../node_modules/helmet/dist/middlewares/x-xss-protection/index.d.ts(1,49): error TS2307: Cannot find module 'http' or its corresponding type declarations.
       ../node_modules/winston-transport/index.d.ts(6,23): error TS2688: Cannot find type definition file for 'node'.
       ../node_modules/winston-transport/index.d.ts(8,25): error TS2307: Cannot find module 'stream' or its corresponding type declarations.
       ../node_modules/winston/index.d.ts(4,23): error TS2688: Cannot find type definition file for 'node'.
       ../node_modules/winston/index.d.ts(6,31): error TS2307: Cannot find module 'stream' or its corresponding type declarations.
       ../node_modules/winston/index.d.ts(127,28): error TS2503: Cannot find namespace 'NodeJS'.
       ../node_modules/winston/index.d.ts(179,34): error TS2503: Cannot find namespace 'NodeJS'.
       ../node_modules/winston/lib/winston/config/index.d.ts(4,23): error TS2688: Cannot find type definition file for 'node'.
       ../node_modules/winston/lib/winston/transports/index.d.ts(4,23): error TS2688: Cannot find type definition file for 'node'.
       ../node_modules/winston/lib/winston/transports/index.d.ts(6,21): error TS2307: Cannot find module 'http' or its corresponding type declarations.
       ../node_modules/winston/lib/winston/transports/index.d.ts(31,14): error TS2503: Cannot find namespace 'NodeJS'.
       ../node_modules/winston/lib/winston/transports/index.d.ts(77,13): error TS2503: Cannot find namespace 'NodeJS'.
       src/ts/action/actions.ts(1,16): error TS7016: Could not find a declaration file for module 'wu'. '/tmp/build_9815baf8_/node_modules/wu/dist/wu.js' implicitly has an 'any' type.
         Try `npm install @types/wu` if it exists or add a new declaration (.d.ts) file containing `declare module 'wu';`
       src/ts/action/actions.ts(69,40): error TS7006: Parameter 'child' implicitly has an 'any' type.
       src/ts/caches/postgres.ts(1,16): error TS7016: Could not find a declaration file for module 'pg'. '/tmp/build_9815baf8_/node_modules/pg/lib/index.js' implicitly has an 'any' type.
         Try `npm install @types/pg` if it exists or add a new declaration (.d.ts) file containing `declare module 'pg';`
       src/ts/caches/postgres.ts(8,23): error TS7016: Could not find a declaration file for module 'uuid'. '/tmp/build_9815baf8_/node_modules/uuid/dist/index.js' implicitly has an 'any' type.
         Try `npm install @types/uuid` if it exists or add a new declaration (.d.ts) file containing `declare module 'uuid';`
       src/ts/config.ts(179,19): error TS2580: Cannot find name 'process'. Do you need to install type definitions for node? Try `npm i @types/node`.
       src/ts/errors/action-execution-error.ts(20,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/action-execution-error.ts(31,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/action-execution-error.ts(49,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/action-execution-error.ts(79,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/action-execution-error.ts(107,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/action-execution-error.ts(139,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/action-execution-error.ts(168,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/action-execution-error.ts(192,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/action-execution-error.ts(211,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/authentication.ts(25,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/authentication.ts(43,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/authentication.ts(52,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/authentication.ts(61,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/authentication.ts(71,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/authentication.ts(81,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/authentication.ts(91,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/game-state-error.ts(11,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/game-state-error.ts(20,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/lobby.ts(22,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/lobby.ts(39,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/lobby.ts(49,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/registration.ts(19,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/errors/validation.ts(15,11): error TS2339: Property 'captureStackTrace' does not exist on type 'ErrorConstructor'.
       src/ts/games/cards/card.ts(1,23): error TS7016: Could not find a declaration file for module 'uuid'. '/tmp/build_9815baf8_/node_modules/uuid/dist/index.js' implicitly has an 'any' type.
         Try `npm install @types/uuid` if it exists or add a new declaration (.d.ts) file containing `declare module 'uuid';`
       src/ts/games/cards/card.ts(2,16): error TS7016: Could not find a declaration file for module 'wu'. '/tmp/build_9815baf8_/node_modules/wu/dist/wu.js' implicitly has an 'any' type.
         Try `npm install @types/wu` if it exists or add a new declaration (.d.ts) file containing `declare module 'wu';`
       src/ts/games/cards/card.ts(105,17): error TS7006: Parameter 'part' implicitly has an 'any' type.
       src/ts/games/cards/decks.ts(1,16): error TS7016: Could not find a declaration file for module 'wu'. '/tmp/build_9815baf8_/node_modules/wu/dist/wu.js' implicitly has an 'any' type.
         Try `npm install @types/wu` if it exists or add a new declaration (.d.ts) file containing `declare module 'wu';`
       src/ts/games/cards/decks.ts(142,49): error TS7006: Parameter 'template' implicitly has an 'any' type.
       src/ts/games/cards/decks.ts(144,24): error TS7006: Parameter 'template' implicitly has an 'any' type.
       src/ts/games/cards/play.ts(1,23): error TS7016: Could not find a declaration file for module 'uuid'. '/tmp/build_9815baf8_/node_modules/uuid/dist/index.js' implicitly has an 'any' type.
         Try `npm install @types/uuid` if it exists or add a new declaration (.d.ts) file containing `declare module 'uuid';`
       src/ts/games/cards/sources/builtIn.ts(3,19): error TS7016: Could not find a declaration file for module 'json5'. '/tmp/build_9815baf8_/node_modules/json5/lib/index.js' implicitly has an 'any' type.
         Try `npm install @types/json5` if it exists or add a new declaration (.d.ts) file containing `declare module 'json5';`
       src/ts/games/cards/sources/builtIn.ts(4,32): error TS2307: Cannot find module 'fs' or its corresponding type declarations.
       src/ts/games/cards/sources/builtIn.ts(6,23): error TS2307: Cannot find module 'path' or its corresponding type declarations.
       src/ts/games/cards/sources/many-decks.ts(2,25): error TS7016: Could not find a declaration file for module 'generic-pool'. '/tmp/build_9815baf8_/node_modules/generic-pool/index.js' implicitly has an 'any' type.
         Try `npm install @types/generic-pool` if it exists or add a new declaration (.d.ts) file containing `declare module 'generic-pool';`
       src/ts/games/cards/sources/many-decks.ts(11,19): error TS7016: Could not find a declaration file for module 'json5'. '/tmp/build_9815baf8_/node_modules/json5/lib/index.js' implicitly has an 'any' type.
         Try `npm install @types/json5` if it exists or add a new declaration (.d.ts) file containing `declare module 'json5';`
       src/ts/games/cards/sources/many-decks.ts(148,25): error TS7006: Parameter '_' implicitly has an 'any' type.
       src/ts/games/game.ts(1,16): error TS7016: Could not find a declaration file for module 'wu'. '/tmp/build_9815baf8_/node_modules/wu/dist/wu.js' implicitly has an 'any' type.
         Try `npm install @types/wu` if it exists or add a new declaration (.d.ts) file containing `declare module 'wu';`
       src/ts/games/game.ts(171,32): error TS7006: Parameter 't' implicitly has an 'any' type.
       src/ts/games/game.ts(183,14): error TS7031: Binding element 'id' implicitly has an 'any' type.
       src/ts/games/game.ts(183,18): error TS7031: Binding element '_' implicitly has an 'any' type.
       src/ts/games/game.ts(187,19): error TS7031: Binding element '_' implicitly has an 'any' type.
       src/ts/games/game.ts(187,22): error TS7031: Binding element 'user' implicitly has an 'any' type.
       src/ts/games/game.ts(188,16): error TS7031: Binding element 'id' implicitly has an 'any' type.
       src/ts/games/game.ts(188,20): error TS7031: Binding element '_' implicitly has an 'any' type.
       src/ts/games/game.ts(201,31): error TS7006: Parameter 'id' implicitly has an 'any' type.
       src/ts/games/game.ts(206,34): error TS2345: Argument of type 'Set<unknown>' is not assignable to parameter of type 'Set<string>'.
       src/ts/games/game.ts(256,36): error TS7006: Parameter 'id' implicitly has an 'any' type.
       src/ts/games/game.ts(262,51): error TS2345: Argument of type 'Set<unknown>' is not assignable to parameter of type 'Set<string>'.
       src/ts/games/game.ts(267,9): error TS2345: Argument of type 'Set<unknown>' is not assignable to parameter of type 'Set<string>'.
       src/ts/games/game/round.ts(1,16): error TS7016: Could not find a declaration file for module 'wu'. '/tmp/build_9815baf8_/node_modules/wu/dist/wu.js' implicitly has an 'any' type.
         Try `npm install @types/wu` if it exists or add a new declaration (.d.ts) file containing `declare module 'wu';`
       src/ts/games/game/round.ts(77,40): error TS7006: Parameter 'play' implicitly has an 'any' type.
       src/ts/games/game/round.ts(165,40): error TS7006: Parameter 'play' implicitly has an 'any' type.
       src/ts/games/game/round.ts(260,40): error TS7006: Parameter 'play' implicitly has an 'any' type.
       src/ts/games/game/round.ts(288,62): error TS7006: Parameter 'play' implicitly has an 'any' type.
       src/ts/games/game/round.ts(291,27): error TS2345: Argument of type 'unknown[]' is not assignable to parameter of type 'string[]'.
         Type 'unknown' is not assignable to type 'string'.
       src/ts/games/game/round.ts(468,32): error TS7006: Parameter 'p' implicitly has an 'any' type.
       src/ts/games/game/round.ts(470,5): error TS2322: Type 'Set<unknown> | null' is not assignable to type 'Set<string> | null'.
         Type 'Set<unknown>' is not assignable to type 'Set<string>'.
           Type 'unknown' is not assignable to type 'string'.
       src/ts/games/game/round.ts(474,40): error TS7006: Parameter 'play' implicitly has an 'any' type.
       src/ts/index.ts(1,24): error TS7016: Could not find a declaration file for module 'body-parser'. '/tmp/build_9815baf8_/node_modules/body-parser/index.js' implicitly has an 'any' type.
         Try `npm install @types/body-parser` if it exists or add a new declaration (.d.ts) file containing `declare module 'body-parser';`
       src/ts/index.ts(2,58): error TS7016: Could not find a declaration file for module 'express'. '/tmp/build_9815baf8_/node_modules/express/index.js' implicitly has an 'any' type.
         Try `npm install @types/express` if it exists or add a new declaration (.d.ts) file containing `declare module 'express';`
       src/ts/index.ts(5,16): error TS7016: Could not find a declaration file for module 'express-ws'. '/tmp/build_9815baf8_/node_modules/express-ws/index.js' implicitly has an 'any' type.
         Try `npm install @types/express-ws` if it exists or add a new declaration (.d.ts) file containing `declare module 'express-ws';`
       src/ts/index.ts(6,32): error TS2307: Cannot find module 'fs' or its corresponding type declarations.
       src/ts/index.ts(9,19): error TS7016: Could not find a declaration file for module 'json5'. '/tmp/build_9815baf8_/node_modules/json5/lib/index.js' implicitly has an 'any' type.
         Try `npm install @types/json5` if it exists or add a new declaration (.d.ts) file containing `declare module 'json5';`
       src/ts/index.ts(10,30): error TS7016: Could not find a declaration file for module 'source-map-support'. '/tmp/build_9815baf8_/node_modules/source-map-support/source-map-support.js' implicitly has an 'any' type.
         Try `npm install @types/source-map-support` if it exists or add a new declaration (.d.ts) file containing `declare module 'source-map-support';`
       src/ts/index.ts(11,16): error TS7016: Could not find a declaration file for module 'wu'. '/tmp/build_9815baf8_/node_modules/wu/dist/wu.js' implicitly has an 'any' type.
         Try `npm install @types/wu` if it exists or add a new declaration (.d.ts) file containing `declare module 'wu';`
       src/ts/index.ts(33,1): error TS2580: Cannot find name 'process'. Do you need to install type definitions for node? Try `npm i @types/node`.
       src/ts/index.ts(33,43): error TS7006: Parameter 'error' implicitly has an 'any' type.
       src/ts/index.ts(37,1): error TS2580: Cannot find name 'process'. Do you need to install type definitions for node? Try `npm i @types/node`.
       src/ts/index.ts(37,44): error TS7006: Parameter 'reason' implicitly has an 'any' type.
       src/ts/index.ts(37,52): error TS7006: Parameter 'promise' implicitly has an 'any' type.
       src/ts/index.ts(46,22): error TS2580: Cannot find name 'process'. Do you need to install type definitions for node? Try `npm i @types/node`.
npm       src/ts/index.ts(81,34): error TS7006: Parameter 'req' implicitly has an 'any' type.
 ERR! code ELIFECYCLE
npm ERR!        src/ts/index.ts(81,39): error TS7006: Parameter 'res' implicitly has an 'any' type.
errno 2
       src/ts/index.ts(83,32): error TS7006: Parameter 'req' implicitly has an 'any' type.
       src/ts/index.ts(83,37): error TS7006: Parameter 'res' implicitly has an 'any' type.
       src/ts/index.ts(91,33): error TS7006: Parameter 'req' implicitly has an 'any' type.
       src/ts/index.ts(91,38): error TS7006: Parameter 'res' implicitly has an 'any' type.
       src/ts/index.ts(107,33): error TS7006: Parameter 'req' implicitly has an 'any' type.
       src/ts/index.ts(107,38): error TS7006: Parameter 'res' implicitly has an 'any' type.
       src/ts/index.ts(126,43): error TS7006: Parameter 'req' implicitly has an 'any' type.
       src/ts/index.ts(126,48): error TS7006: Parameter 'res' implicitly has an 'any' type.
       src/ts/index.ts(139,46): error TS7006: Parameter 'u' implicitly has an 'any' type.
       src/ts/index.ts(186,41): error TS7006: Parameter 'socket' implicitly has an 'any' type.
       src/ts/index.ts(186,49): error TS7006: Parameter 'req' implicitly has an 'any' type.
       src/ts/index.ts(191,34): error TS7006: Parameter 'req' implicitly has an 'any' type.
       src/ts/index.ts(191,39): error TS7006: Parameter 'res' implicitly has an 'any' type.
       src/ts/index.ts(267,3): error TS2580: Cannot find name 'process'. Do you need to install type definitions for node? Try `npm i @types/node`.
       src/ts/secret.ts(1,20): error TS2307: Cannot find module 'crypto' or its corresponding type declarations.
npm ERR! @massivedecks/server@2.0.0 build: `npx tsc`
npm ERR! Exit status 2
npm ERR! 
npm ERR! Failed at the @massivedecks/server@2.0.0 build script.
npm        src/ts/socket-manager.ts(1,23): error TS7016: Could not find a declaration file for module 'ws'. '/tmp/build_9815baf8_/node_modules/ws/index.js' implicitly has an 'any' type.
ERR! This is probably not a problem with npm. There is likely additional logging output above.
         Try `npm install @types/ws` if it exists or add a new declaration (.d.ts) file containing `declare module 'ws';`
       src/ts/store/in-memory.ts(1,23): error TS7016: Could not find a declaration file for module 'uuid'. '/tmp/build_9815baf8_/node_modules/uuid/dist/index.js' implicitly has an 'any' type.
         Try `npm install @types/uuid` if it exists or add a new declaration (.d.ts) file containing `declare module 'uuid';`
       src/ts/store/in-memory.ts(2,16): error TS7016: Could not find a declaration file for module 'wu'. '/tmp/build_9815baf8_/node_modules/wu/dist/wu.js' implicitly has an 'any' type.
         Try `npm install @types/wu` if it exists or add a new declaration (.d.ts) file containing `declare module 'wu';`
       src/ts/store/in-memory.ts(14,16): error TS2665: Invalid module name in augmentation. Module 'wu' resolves to an untyped module at '/tmp/build_9815baf8_/node_modules/wu/dist/wu.js', which cannot be augmented.
       src/ts/store/in-memory.ts(61,17): error TS7031: Binding element '_' implicitly has an 'any' type.
       src/ts/store/in-memory.ts(61,22): error TS7031: Binding element 'lobby' implicitly has an 'any' type.
       src/ts/store/in-memory.ts(62,14): error TS7031: Binding element 'gameCode' implicitly has an 'any' type.
       src/ts/store/in-memory.ts(62,26): error TS7031: Binding element 'lobby' implicitly has an 'any' type.
       src/ts/store/in-memory.ts(141,12): error TS7006: Parameter 'u' implicitly has an 'any' type.
       src/ts/store/postgres.ts(1,16): error TS7016: Could not find a declaration file for module 'pg'. '/tmp/build_9815baf8_/node_modules/pg/lib/index.js' implicitly has an 'any' type.
         Try `npm install @types/pg` if it exists or add a new declaration (.d.ts) file containing `declare module 'pg';`
       src/ts/store/postgres.ts(2,23): error TS7016: Could not find a declaration file for module 'uuid'. '/tmp/build_9815baf8_/node_modules/uuid/dist/index.js' implicitly has an 'any' type.
         Try `npm install @types/uuid` if it exists or add a new declaration (.d.ts) file containing `declare module 'uuid';`
       src/ts/task/start-game.ts(1,16): error TS7016: Could not find a declaration file for module 'wu'. '/tmp/build_9815baf8_/node_modules/wu/dist/wu.js' implicitly has an 'any' type.
         Try `npm install @types/wu` if it exists or add a new declaration (.d.ts) file containing `declare module 'wu';`
       src/ts/task/start-game.ts(21,27): error TS7006: Parameter 'deck' implicitly has an 'any' type.
npm ERR! A complete log of this run can be found in:
npm ERR!     /tmp/npmcache.HFFZc/_logs/2020-10-26T09_20_26_149Z-debug.log
       src/ts/timeout/finished-playing.ts(1,16): error TS7016: Could not find a declaration file for module 'wu'. '/tmp/build_9815baf8_/node_modules/wu/dist/wu.js' implicitly has an 'any' type.
         Try `npm install @types/wu` if it exists or add a new declaration (.d.ts) file containing `declare module 'wu';`
       src/ts/timeout/finished-playing.ts(16,50): error TS7006: Parameter 'play' implicitly has an 'any' type.
       src/ts/timeout/user-disconnect.ts(5,16): error TS7016: Could not find a declaration file for module 'wu'. '/tmp/build_9815baf8_/node_modules/wu/dist/wu.js' implicitly has an 'any' type.
         Try `npm install @types/wu` if it exists or add a new declaration (.d.ts) file containing `declare module 'wu';`
       src/ts/user/token.ts(1,17): error TS7016: Could not find a declaration file for module 'jsonwebtoken'. '/tmp/build_9815baf8_/node_modules/jsonwebtoken/index.js' implicitly has an 'any' type.
         Try `npm install @types/jsonwebtoken` if it exists or add a new declaration (.d.ts) file containing `declare module 'jsonwebtoken';`
       src/ts/util/postgres.ts(1,16): error TS7016: Could not find a declaration file for module 'pg'. '/tmp/build_9815baf8_/node_modules/pg/lib/index.js' implicitly has an 'any' type.
         Try `npm install @types/pg` if it exists or add a new declaration (.d.ts) file containing `declare module 'pg';`
npm ERR! weird error 2
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! massivedecks@2.0.0 buildServer: `npm explore @massivedecks/server -- npm run build`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the massivedecks@2.0.0 buildServer script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

**Comment by CookieCoder15** [comment]
I'm trying to work on this again, looking back over the logs these two lines are the ones I want to deal with first:

```
error TS6047: Argument for '--target' option must be 'ES3', 'ES5', or 'ES6'.
error TS6046: Argument for '--module' option must be 'commonjs', 'amd', 'system' or 'umd'.
```

I'm assuming that's referring to /server/tsconfig.json:

```
"compilerOptions": {
    "target": "esnext",
    "module": "esnext",
```

Which options from the ones Heroku suggested above should I use to replace esnext?

**Comment by Lattyware** [comment]
All of those errors are because it is installing an old version of the
typescript compiler that doesn't support the newer options. None of those
old options will work for the application, as the project uses modern
features.

The answer will be getting Heroku to correctly install and use the version
of the typescript compiler specified in `package.json` so it recognises the
newer options.

**Comment by CookieCoder15** [comment]
Yeah, I realised I was wrong and deleted my comment shortly after sending. (Forgot about email replies. Oops!) Thanks for the clarification into the issue, I'll have a look at it soon since I host using Heroku and would like to get this working again.

**Comment by lhvy** [comment]
Tested again, and appears to be fixed as of 82b95132260d503584e949de7ca012240d41a683
